core-bundle/src/EventListener/PrettyErrorScreenListener.php

```diff
@@ -250,4 +247,25 @@ private function getStatusCodeForException(\Throwable $exception): int
 
         return 500;
     }
+
+    private function findErrorPage(int $type, Request $request): PageModel|null
+    {
+        $pageModel = $request->attributes->get('pageModel');
+
+        if (!$pageModel instanceof PageModel) {
+            $rootRequest = Request::create('http://'.$request->getHost());
+            $rootRequest->headers->set('Accept-Language', $rootRequest->headers->get('Accept-Language'));
+            $parameters = $this->requestMatcher->matchRequest($request);
```

Comment:
Right now it works neither with nor without legacy routing. I think this needs to be changed?

```suggestion
            $parameters = $this->requestMatcher->matchRequest($rootRequest);
```